pkgs/development/tools/misc/texlab/default.nix

```diff
@@ -2,41 +2,52 @@
 , stdenv
 , rustPlatform
 , fetchFromGitHub
+, help2man
 , installShellFiles
 , libiconv
 , Security
 , CoreServices
 , nix-update-script
 }:
 
-rustPlatform.buildRustPackage rec {
+let
+  isCross = stdenv.hostPlatform != stdenv.buildPlatform;
+in rustPlatform.buildRustPackage rec {
   pname = "texlab";
-  version = "4.2.0";
+  version = "4.2.1";
 
   src = fetchFromGitHub {
     owner = "latex-lsp";
     repo = pname;
-    rev = "v${version}";
-    sha256 = "sha256-oYM+OAYjQ8aNAryg0Cthj14BsxMFnOtz38XdUQZZolk=";
+    rev = "refs/tags/v${version}";
+    sha256 = "sha256-za3TauhNoxGDphpY615EnTt46HpMgS+sYpBln/twefw=";
   };
 
-  cargoSha256 = "sha256-TDGiqC9eNIJfLTc1R3nvE84rAsVE8jtTaeQbVNMeVgg=";
+  cargoSha256 = "sha256-wppaa3IGOqkFu/1CAp8g+PlPtMWm/7qNpPu0k4/mL3A=";
 
-  outputs = [ "out" "man" ];
+  outputs = [ "out" ] ++ lib.optional (!isCross) "man";
 
-  nativeBuildInputs = [ installShellFiles ];
+  nativeBuildInputs = [ installShellFiles ]
+  ++ lib.optional (!isCross) help2man;
 
-  buildInputs = lib.optionals stdenv.isDarwin [ libiconv Security CoreServices ];
+  buildInputs = lib.optionals stdenv.isDarwin [
+    libiconv
+    Security
+    CoreServices
+  ];
 
   postInstall = ''
-    installManPage texlab.1
-
     # Remove generated dylib of human_name dependency. TexLab statically
     # links to the generated rlib and doesn't reference the dylib. I
     # couldn't find any way to prevent building this by passing cargo flags.
     # See https://github.com/djudd/human-name/blob/master/Cargo.toml#L43
     rm "$out/lib/libhuman_name${stdenv.hostPlatform.extensions.sharedLibrary}"
     rmdir "$out/lib"
+  '' + lib.optionalString (!isCross) ''
```

Comment:
```suggestion
  ''
  # When we cross compile we cannot run the output executable to generate
  # the man page.
  + lib.optionalString (!isCross) ''
```

Reply:
Oh thanks that's a good point! I had to do this for another package, so I forgot to add a comment here :sweat_smile: